runtime/src/bank.rs

```diff
@@ -2764,6 +2765,33 @@ impl Bank {
         }
     }
 
+    #[allow(dead_code)]
+    fn hash_rewards_into_partitions(
+        stake_rewards: &StakeRewards,
+        rewarded_epoch: Epoch,
+        rewards: u64,
+        num_partitions: usize,
+        thread_pool: &ThreadPool,
+    ) -> Vec<StakeRewards> {
+        let seed = rewarded_epoch ^ rewards;
+
+        let mut result = vec![vec![]; num_partitions];
+
+        thread_pool.install(|| {
+            let buckets = stake_rewards
+                .par_iter()
+                .map(|reward| {
+                    address_to_partition(num_partitions, seed, &reward.stake_pubkey)
+                })
+                .collect::<Vec<_>>();
+
+            for (bucket, reward) in std::iter::zip(buckets, stake_rewards) {
+                result[bucket].push(reward.clone());
+            }
+            result
+        })
```

Comment:
maybe we aim for 1/2 of our ideal 4096 accounts/partition, under the assumption that it will be difficult/unlikely to get more than 2x the accounts mapping to one partition as compared to the average. This doesn't help the attack issue, but it helps the normal case.

Reply:
i think the `4096` entry partition size can just be dropped. it's pretty arbitrary. i have stats for hash-to-bucket with various hash functions and i think the distribution per-epoch is sufficiently fair and over time, even more so. unintentional degenerate cases don't seem any more risky here than they are in any of the other places that we use deterministic, random position-assignment-like operations.

Reply:
![image](https://github.com/solana-labs/solana/assets/75863576/b3b33f8c-3d20-40f3-8fd5-9d95b01e7506)
Here's a result of running 10k attempts with a different seed each time.
looks uniform.

Reply:
4096 is the # we use to determine how many partitions to divide into. Obviously we would now just write all the stake accounts that has into that bin, which should average 4096.

Reply:
> 4096 is the # we use to determine how many partitions to divide into. Obviously we would now just write all the stake accounts that has into that bin, which should average 4096.

ah ok, that must've changed since my last read. if it's just a target, that's fine by me.

> Here's a result of running 10k attempts with a different seed each time.
> looks uniform.

your stats look on par with mine, ~10% average spread in bucket population